src/context/ModalProvider/types.ts

```diff
@@ -0,0 +1,57 @@
+import type { FC } from 'react'
+import type { BackupPassphraseModalProps } from 'components/Layout/Header/NavBar/Native/BackupPassphraseModal/BackupPassphraseModal'
+import type { AssetSearchModalProps } from 'components/Modals/AssetSearch/AssetSearchModal'
+import type { FiatRampsModalProps } from 'components/Modals/FiatRamps/FiatRampsModal'
+import type { NftModalProps } from 'components/Modals/Nfts/NftModal'
+import type { PopupWindowModalProps } from 'components/Modals/PopupWindowModal'
+import type { QrCodeModalProps } from 'components/Modals/QrCode/QrCode'
+import type { ReceivePropsType } from 'components/Modals/Receive/Receive'
+import type { SendModalProps } from 'components/Modals/Send/Send'
+
+import type { CLOSE_MODAL, OPEN_MODAL } from './constants'
+
+export type Modals = {
+  receive: ({ asset, accountId }: ReceivePropsType) => JSX.Element
+  qrCode: ({ assetId, accountId }: QrCodeModalProps) => JSX.Element
+  send: ({ assetId, accountId, input }: SendModalProps) => JSX.Element
+  fiatRamps: FC<FiatRampsModalProps>
+  settings: () => JSX.Element
```

Comment:
All modals are functional components and could be typed as such e.g
```suggestion
  settings: FC
```